benchmark/merlin_hashtable_benchmark.cc.cu

```diff
@@ -169,90 +222,99 @@ void test_main(const size_t dim,
-
-  int32_t capacity = static_cast<int32_t>(init_capacity / (1024 * 1024));
-  size_t hmem4values =
-      init_capacity * options.dim * sizeof(V) / (1024 * 1024 * 1024);
-  hmem4values = hmem4values < hbm4values ? 0 : (hmem4values - hbm4values);
-  float insert_or_assign_tput = key_num_per_op /
-                                diff_insert_or_assign.getResult() /
-                                (1024 * 1024 * 1024.0);
-  float find_tput =
-      key_num_per_op / diff_find.getResult() / (1024 * 1024 * 1024.0f);
-  float find_or_insert_tput =
-      key_num_per_op / diff_find_or_insert.getResult() / (1024 * 1024 * 1024.0f);
-  float assign_tput =
-      key_num_per_op / diff_assign.getResult() / (1024 * 1024 * 1024.0f);
-
-  cout << "|" << rep(1) << setw(3) << setfill(' ') << options.dim << " "
-       << "|" << rep(15) << setw(5) << setfill(' ') << capacity << " "
-       << "|" << rep(9) << setw(3) << setfill(' ') << hbm4values << " /"
-       << setw(3) << setfill(' ') << hmem4values << " "
-       << "|" << rep(8) << fixed << setprecision(2) << load_factor << " "
-       << "|" << rep(12) << fixed << setprecision(3) << insert_or_assign_tput
-       << " "
-       << "|" << rep(2) << fixed << setprecision(3) << find_tput << " "
-       << "|" << rep(10) << fixed << setprecision(3) << find_or_insert_tput
-       << " "
-       << "|" << rep(2) << fixed << setprecision(3) << assign_tput << " |"
-       << endl;
+  create_keys_for_hitrate<K, M>(h_keys, h_metas, key_num_per_op, hitrate,
+                                Hit_Mode::last_insert, start, true /*reset*/);
+  CUDA_CHECK(cudaMemcpy(d_keys, h_keys, key_num_per_op * sizeof(K),
+                        cudaMemcpyHostToDevice));
+  CUDA_CHECK(cudaMemcpy(d_metas, h_metas, key_num_per_op * sizeof(M),
+                        cudaMemcpyHostToDevice));
+  auto timer = benchmark::Timer<double>();
+  switch (api) {
+    case API_Select::find: {
+      timer.start();
+      table->find(key_num_per_op, d_keys, d_vectors, d_found, d_metas, stream);
+      CUDA_CHECK(cudaStreamSynchronize(stream));
+      timer.end();
+      break;
+    }
+    case API_Select::insert_or_assign: {
+      timer.start();
+      table->insert_or_assign(key_num_per_op, d_keys, d_vectors, d_metas,
+                              stream);
+      CUDA_CHECK(cudaStreamSynchronize(stream));
+      timer.end();
+      break;
+    }
+    case API_Select::find_or_insert: {
+      timer.start();
+      table->find_or_insert(key_num_per_op, d_keys, d_vectors, d_metas, stream);
+      CUDA_CHECK(cudaStreamSynchronize(stream));
+      timer.end();
+      break;
+    }
+    case API_Select::assign: {
+      timer.start();
+      table->assign(key_num_per_op, d_keys, d_def_val, d_metas, stream);
+      CUDA_CHECK(cudaStreamSynchronize(stream));
+      timer.end();
+      break;
+    }
+    case API_Select::insert_and_evict: {
+      timer.start();
+      table->insert_and_evict(key_num_per_op, d_keys, d_vectors, d_metas,
```

Comment:
The first calling will not be accurate, it should be called once before real timing.